tests/conftest.py

```diff
@@ -910,6 +912,7 @@ def tier_marks_name():
 @pytest.fixture(scope='function', autouse=True)
 def health_checker(request, tier_marks_name):
     node = request.node
+    request.addfinalizer(ceph_health_check)
```

Comment:
ceph_health_check doesn't throw an exception, just returns bool, so here, it will have no effect in case ceph health if not OK.
A solution would be to create a finalizer function that calls ceph_health_check and asserts its result

Reply:
ceph_health_check return what is being returned from ceph_health_check_base:
```
def ceph_health_check_base(namespace=None):
    """
    Exec `ceph health` cmd on tools pod to determine health of cluster.

    Args:
        namespace (str): Namespace of OCS
            (default: config.ENV_DATA['cluster_namespace'])

    Raises:
        CephHealthException: If the ceph health returned is not HEALTH_OK
        CommandFailed: If the command to retrieve the tools pod name or the
            command to get ceph health returns a non-zero exit code
    Returns:
        boolean: True if HEALTH_OK

    """
```
Which means a CephHealthException is thrown when the state is not OK

Reply:
maybe I'm wrong but IIUC, ocs_ci.utility.retry.retry catches the exception raised by ocs_ci.utility.utils.ceph_health_check_base and returns False in case the cluster health is not OK

Reply:
```
def retry(exception_to_check, tries=4, delay=3, backoff=2):
    """
    """
    def deco_retry(f):

        @wraps(f)
        def f_retry(*args, **kwargs):
            mtries, mdelay = tries, delay
            while mtries > 1:   <<--- while there are still tries, it will catch the exception and the call is ** return f(*args, **kwargs) **
                try:
                    return f(*args, **kwargs)
                except exception_to_check as e:
                    logger.warning("%s, Retrying in %d seconds..." % (str(e), mdelay))
                    time.sleep(mdelay)
                    mtries -= 1
                    mdelay *= backoff
            return f(*args, **kwargs)  <<--- when no tries left, it will call it once more with no try: except: which means it will trow the exception 

        return f_retry

    return deco_retry
```

Reply:
@RazTamir Thanks! just validated it myself and you are right :)